### PR TITLE
Add checkpointing in missing methods

### LINKED ISSUE #83: Checkpoint deposit and global reward in `alterClaimer` and `alterDelegatee`
#### Description

- Add test based to verify shares are not inflated
- Add something like below 
```
  function _alterClaimer(
    Deposit storage deposit,
    DepositIdentifier _depositId,
    address _newClaimer
  ) internal virtual {
    _revertIfAddressZero(_newClaimer);

+   _checkpointGlobalReward();
+   _checkpointReward(deposit);
```

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description

- Checkpoint global reward and deposit when altering a delegatee and claimer

Closes #83

### CODE REVIEW

File: test/GovernanceStaker.t.sol [hunk 1]
```diff
@@ -2764,6 +2764,58 @@ contract NotifyRewardAmount is GovernanceStakerRewardsTest {
     assertLe(govStaker.scaledRewardRate(), _expectedRewardRate);
   }
 
+  function test_shareManipulationIfRewardsArentCeckpointed() public {
```
Comment (wildmolasses) [inline test/GovernanceStaker.t.sol:2767]:
```suggestion
  function test_shareManipulationIfRewardsArentCheckpointed() public {
```

Comment (wildmolasses) [reply test/GovernanceStaker.t.sol:2767]: should we rename this test to describe that there's no longer share manipulation? also, are there unit tests to write here (e.g. alterDelegatee checkpoints global reward, alterDelegatee checkpoints reward)? defer to you on that

File: test/GovernanceStaker.t.sol [hunk 2]
```diff
@@ -2764,6 +2764,58 @@ contract NotifyRewardAmount is GovernanceStakerRewardsTest {
...
+    earningPowerCalculator.__setEarningPowerForDelegatee(address(0x1), 0);
+    earningPowerCalculator.__setEarningPowerForDelegatee(address(0x2), 500e18);
+
+    address _doe = makeAddr("doe");
+    _mintGovToken(_doe, 500e18);
+    _stake(_doe, 500e18, address(0x2));
+
+    address _fox = makeAddr("fox");
+
+    //!audit-ok Step 1, fox deposits with full earning power
```
Comment (wildmolasses) [inline test/GovernanceStaker.t.sol:2777]: might revise these comments, or possibly attribute [CergyK](https://github.com/CergyK) above the test

Comment (alexkeating) [reply test/GovernanceStaker.t.sol]: Updated

**Pull request merged**